### PR TITLE
[OJS][stable-3_3_0] #5697 Enable edit galley, display the form disabled

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

**Review by NateWr — CHANGES_REQUESTED** [review]

**Comment by NateWr** [comment]
The last commit message needs to be `Submodule update ##defstat/<branch>##` in order to get the tests running correctly.

File: plugins/pubIds/doi/templates/doiSuffixEdit.tpl [hunk 1]
```diff
@@ -34,16 +36,23 @@
...
-				{assign var=templatePath value=$pubIdPlugin->getTemplateResource('doiAssignCheckBox.tpl')}
-				{include file=$templatePath pubId="" pubObjectType=$pubObjectType}
+				{if !$formDisabled}
+					<p class="pkp_help">{translate key="plugins.pubIds.doi.editor.canBeAssigned"}</p>
+					{assign var=templatePath value=$pubIdPlugin->getTemplateResource('doiAssignCheckBox.tpl')}
+					{include file=$templatePath pubId="" pubObjectType=$pubObjectType}
+				{else}
+					<p class="pkp_help">{translate key="plugins.pubIds.doi.editor.canBeAssigned.disabled"}</p>
+				{/if}
+
```
Comment (NateWr) [inline plugins/pubIds/doi/templates/doiSuffixEdit.tpl:55]: If the form is disabled, we only want to show a DOI if it is assigned. `$pubIdPlugin->getPubId($pubObject)` will get the DOI that _will be assigned_ when the action is taken. But in such cases, the action can't be taken.

So if the form is disabled show a) the DOI that is assigned to the galley or b) text that says no doi is assigned to the galley.

**Review by NateWr — CHANGES_REQUESTED** [review]

File: controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php [hunk 2]
```diff
@@ -56,20 +56,20 @@ function initialize($request, $template = null) {
...
+			// Add row-level actions
+			import('lib.pkp.classes.linkAction.request.AjaxModal');
+			$this->addAction(new LinkAction(
+				'editGalley',
+				new AjaxModal(
+					$router->url($request, null, null, 'editGalley', null, $actionArgs),
+					__('submission.layout.editGalley'),
+					'modal_edit'
+				),
+				__('grid.action.edit'),
```
[the remark this answers is not in the document]
Comment (defstat) [reply controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php]: Fixed

File: controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php [hunk 3]
```diff
@@ -56,20 +56,20 @@ function initialize($request, $template = null) {
 			$actionArgs = $this->getRequestArgs();
 			$actionArgs['representationId'] = $rowId;
 
+			// Add row-level actions
+			import('lib.pkp.classes.linkAction.request.AjaxModal');
+			$this->addAction(new LinkAction(
+				'editGalley',
+				new AjaxModal(
+					$router->url($request, null, null, 'editGalley', null, $actionArgs),
+					__('submission.layout.editGalley'),
```
[the remark this answers is not in the document]
Comment (defstat) [reply controllers/grid/articleGalleys/ArticleGalleyGridRow.inc.php]: Fixed

File: plugins/pubIds/doi/templates/doiSuffixEdit.tpl (hunk 1 above)
Comment (defstat) [reply plugins/pubIds/doi/templates/doiSuffixEdit.tpl:55]: Changed the locale key to `plugins.pubIds.doi.editor.noDoiAssigned` with the text `No DOI is assigned yet.`. Removed the preview of the DOI

Comment (NateWr) [reply plugins/pubIds/doi/templates/doiSuffixEdit.tpl:57]: Can you add a new locale key that says "No DOI is assigned."? The word "yet" implies that someone can change it but once it's published they can't.

Comment (defstat) [reply plugins/pubIds/doi/templates/doiSuffixEdit.tpl:57]: Changed

**Pull request merged**